nemo/collections/nlp/nm/trainables/dialogue_state_tracking/sgd/sgd_model.py

```diff
@@ -0,0 +1,360 @@
+            "logit_noncat_slot_status": NeuralType(('B', 'T', 'C'), LogitsType()),
+            "non_cat_slot_status_mask": NeuralType(('B'), ChannelType()),
+            "logit_noncat_slot_start": NeuralType(('B', 'T', 'C'), LogitsType()),
+            "logit_noncat_slot_end": NeuralType(('B', 'T', 'C'), LogitsType()),
+        }
+
+    def __init__(self, embedding_dim, schema_emb_processor):
+        """Get logits for elements by conditioning on utterance embedding.
+
+        Args:
+            embedding_dim (int): hidden size of the BERT
+            schema_emb_processor (obj): contains schema embeddings for services and config file
+        """
+        super().__init__()
+
+        # Add a trainable vector for the NONE intent
+        self.none_intent_vector = torch.empty((1, 1, embedding_dim), requires_grad=True).to(self._device)
+        # TODO truncated norm init
+        nn.init.normal_(self.none_intent_vector, std=0.02)
+        self.none_intent_vector = torch.nn.Parameter(self.none_intent_vector).to(self._device)
+
+        self.intent_layer = Logits(1, embedding_dim).to(self._device)
+        self.requested_slots_layer = Logits(1, embedding_dim).to(self._device)
+
+        self.cat_slot_value_layer = Logits(1, embedding_dim).to(self._device)
+
+        # Slot status values: none, dontcare, active.
+        self.cat_slot_status_layer = Logits(3, embedding_dim).to(self._device)
+        self.noncat_slot_layer = Logits(3, embedding_dim).to(self._device)
+
+        # dim 2 for non_categorical slot - to represent start and end position
+        self.noncat_layer1 = nn.Linear(2 * embedding_dim, embedding_dim).to(self._device)
+        self.noncat_activation = F.gelu
+        self.noncat_layer2 = nn.Linear(embedding_dim, 2).to(self._device)
+
+        config = schema_emb_processor.schema_config
+        num_services = len(schema_emb_processor.schemas.services)
+        self.intents_emb = nn.Embedding(num_services, config["MAX_NUM_INTENT"] * embedding_dim)
+        self.cat_slot_emb = nn.Embedding(num_services, config["MAX_NUM_CAT_SLOT"] * embedding_dim)
+        self.cat_slot_value_emb = nn.Embedding(
+            num_services, config["MAX_NUM_CAT_SLOT"] * config["MAX_NUM_VALUE_PER_CAT_SLOT"] * embedding_dim
+        )
+        self.noncat_slot_emb = nn.Embedding(num_services, config["MAX_NUM_NONCAT_SLOT"] * embedding_dim)
+        self.req_slot_emb = nn.Embedding(
+            num_services, (config["MAX_NUM_CAT_SLOT"] + config["MAX_NUM_NONCAT_SLOT"]) * embedding_dim
+        )
+
+        # initialize schema embeddings from the BERT generated embeddings
+        schema_embeddings = schema_emb_processor.get_schema_embeddings()
+        self.intents_emb.weight.data.copy_(
+            torch.from_numpy(np.stack(schema_embeddings['intent_emb']).reshape(num_services, -1))
+        )
+        self.cat_slot_emb.weight.data.copy_(
+            torch.from_numpy(np.stack(schema_embeddings['cat_slot_emb']).reshape(num_services, -1))
+        )
```

Comment:
Coudn't we simply use the schema_embeddings object instead of:
 * creating several local embeddings
 * copying weights from the schema_embeddings

Any particular reason to make it that complex?